src/main/java/io/airlift/slice/Slices.java

```diff
@@ -36,6 +37,8 @@
      */
     public static final Slice EMPTY_SLICE = new Slice();
 
+    // See java.util.ArrayList for an explanation
+    private static final int MAX_ARRAY_SIZE = Integer.MAX_VALUE - 8;
```

Comment:
Add newline after this so that it's clear the comment above is only for this one constant